### PR TITLE
docker: Add dockerfiles for running dcrd nodes.

### PR DESCRIPTION
**Pull request by orthomind**
This is a basic Dockerfile setup to enable running a dcrd node from inside Docker.  Among other things, it is for now missing a way to configure settings via environment variables, and therefore requires dcrd.conf to be provided directly.  This is something I plan to add in a later PR.

I've provided a standard Dockerfile as well as an alpine based option for reduced size.

### CODE REVIEW

**Comment by dajohi** [comment]
I like this.  Can you add docker build instructions to the README and a travis test?

File: README.md [hunk 1]
```diff
@@ -1,9 +1,15 @@
 dcrd
 ====
 
-[![Build Status](https://travis-ci.org/decred/dcrd.png?branch=master)](https://travis-ci.org/decred/dcrd)
-[![ISC License](http://img.shields.io/badge/license-ISC-blue.svg)](http://copyfree.org)
-[![GoDoc](https://img.shields.io/badge/godoc-reference-blue.svg)](http://godoc.org/github.com/decred/dcrd)
+[![Build Status](
```
[the remark this answers is not in the document]
Comment (orthomind) [reply README.md]: Roger.  Was trying to reduce all lines to fit within 80 characters, but will restore.

[the remark this answers is not in the document]
Comment (orthomind) [reply README.md]: Is this better?  Or did you want the other tweaks restored as well?

[the remark this answers is not in the document]
Comment (davecgh) [reply README.md]: Looks good.  I was just talking about the badges.

**Pull request merged**